Add support for ApiPermission#destroy

jamiemtdwyer opened the pull request:
Fixes https://github.com/Shopify/shopify_api/issues/438.

Adds support for ApiPermission#destroy, which allows applications to remotely uninstall themselves.

~Removes the OAuth class as it's now deprecated.~

jamiemtdwyer commented:
Actually, the OAuth#revoke action still seems to be working, so I don't think we want to 🔥it here. Maybe we can just add a comment to let folks know that it is deprecated in favour of ApiPermission#revoke.

jtgrenz commented on lib/shopify_api/resources/o_auth.rb:
```diff
@@ -1,3 +1,8 @@
+# frozen_string_literal: true
+
+# This resource is deprecated and will likely be removed in a future version of this gem.
+# Use ShopifyAPI::ApiPermission.destroy instead
```
You probably want to log this when someone uses the deprecated method.

```ruby
def self.revoke
  warn "[DEPRICATION] ShopifyAPI::OAuth is deprecated and will likely be removed in a future version of this gem. Use ShopifyAPI::ApiPermission.destroy instead"
  
  delete(:revoke)
end
```
maybe?

jamiemtdwyer commented:
Added the kernel warning as @jtgrenz suggested. I've also changed the class method on ApiPermission from `self.revoke` to `self.destroy` to be more descriptive.
